### PR TITLE
Get request_stack for Symfony 3

### PR DESCRIPTION
**Pull request by schauer**
I am targeting this branch (3.x), because of a bug fix.

## Changelog

```markdown
### Fixed
- deprecation message about the "request" service from the admin controller
```
## Subject

Use the `request_stack` when available, by using `CRUDController::getRequest()`

### CODE REVIEW

File: Controller/SiteAdminController.php [hunk 1]
```diff
@@ -30,13 +31,13 @@ class SiteAdminController extends Controller
      * @throws NotFoundHttpException
      * @throws AccessDeniedException
      */
-    public function snapshotsAction()
+    public function snapshotsAction(Request $request)
```
Comment (greg0ire) [inline Controller/SiteAdminController.php]: This is a BC-break for extending classes

**Comment by greg0ire** [comment]
> because of a bug fix

How is this a bug?

File: Controller/SiteAdminController.php (hunk 1 above)
Comment (Soullivaneuh) [reply Controller/SiteAdminController.php]: I think `CRUDController` of admin-bundle handles the request with a different maner.

Comment (greg0ire) [reply Controller/SiteAdminController.php]: https://github.com/sonata-project/SonataAdminBundle/blob/c39403f1b801cf2d40d916d2cd00daf753adeec9/Controller/CRUDController.php#L948-L955

**Comment by greg0ire** [comment]
> How is this a bug?

Answering my own question: I think this might fix a deprecation notice

**Comment by Soullivaneuh** [comment]
@greg0ire Indeed. This is not really a bug, but the PR is a patch. :wink:

**Comment by greg0ire** [comment]
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already pushed. Don't forget the `--force` option otherwise git will try to merge both things together.

**Comment by greg0ire** [comment]
You should probably copy/paste this: https://github.com/sonata-project/SonataAdminBundle/blob/c39403f1b801cf2d40d916d2cd00daf753adeec9/Controller/CRUDController.php#L948-L955

**Comment by greg0ire** [comment]
TODO:

- [ ] copy / paste the method I asked you to copy / paste
- [ ] amend your commit message to reflect what you are doing now.

**Comment by schauer** [comment]
The requested changes were realized. Is everything correct now?

**Comment by greg0ire** [comment]
Please format your commit message according to our rules : 

- The commit message subject must be less than 50 characters and tell us *what* you did.
- The commit message body should tell us *why* you did it. It is optional but highly recommended.

Bad example :

```
Fixed bug #989 by removing call to defraculate()
```

Good example:

```
Remove call to defraculate()

Calling this function caused a bug because it interferes 
with calls to getSchmeckles().
Fixes #989
```

More details in CONTRIBUTING.md

File: Controller/SiteAdminController.php [hunk 2]
```diff
@@ -63,4 +64,16 @@ public function snapshotsAction()
             'object' => $object,
         ));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequest()
```
Comment (greg0ire) [inline Controller/SiteAdminController.php]: Please make the method final. Not sure why we didn't do it in the piece of code you copied from.

Comment (schauer) [reply Controller/SiteAdminController.php]: The class \Sonata\PageBundle\Controller\SiteAdminController extends the class \Sonata\AdminBundle\Controller\CRUDController. The class \Sonata\AdminBundle\Controller\CRUDController contains the getRequest() I have to copy. I do not know why I had to copy the method at all.

I will do the method final and try to improve my commit message.

Comment (greg0ire) [reply Controller/SiteAdminController.php]: Oooooh I did not notice that. Ok then, just remove it, and sorry for the trouble!

**Comment by greg0ire** [comment]
I updated your changelog and your PR subject.

File: Controller/SiteAdminController.php (hunk 2 above)
Comment (greg0ire) [reply Controller/SiteAdminController.php]: `CRUDController` is aliased to `Controller`, that's probably why I did not notice that.

**Pull request merged**